tools/ResourceMetaSchema.json

```diff
@@ -1,108 +1,200 @@
+  "$schema": "http://json-schema.org/draft-04/schema#",
+  "title": "Azure Resource Meta-schema",
+      
+  "allOf": [
+    {
+      "$ref": "http://json-schema.org/draft-04/schema#"
+    },
+    {
+      "$ref": "#/definitions/resourceMetaSchema"
+    }
+  ],
 
-    "allOf": [
-        {
-            "$ref": "http://json-schema.org/draft-04/schema#"
+  "definitions": {
+    "resourceMetaSchema": {
+      "type": "object",
+      "properties": {
+        "id": { "type": "string" },
+        "$schema": {
+          "type": "string",
+          "format": "uri"
         },
-        {
-            "$ref": "#/definitions/resourceMetaSchema"
-        }
-    ],
-
-    "definitions": {
-        "resourceMetaSchema": {
-            "title": "Azure Resource Meta-schema",
-            "type": "object",
-            "properties": {
-                "id": { "type": "string" },
-                "$schema": {
-                    "type": "string",
-                    "format": "uri"
-                },
-                "title": { "type": "string" },
-                "description": { "$ref": "#/definitions/description" },
-                "resourceDefinitions": {
-                    "type": "object",
-                    "minProperties": 1,
-                    "additionalProperties": { "$ref": "#/definitions/resource" }
-                },
-                "definitions": {
-                    "type": "object",
-                    "minProperties": 1,
-                    "additionalProperties": { "$ref": "http://json-schema.org/draft-04/schema#" }
-                }
-            },
-            "required": [ "$schema", "title", "description", "resourceDefinitions" ],
-            "additionalProperties": false
+        "title": { "type": "string" },
+        "description": { "$ref": "#/definitions/description" },
+        "resourceDefinitions": {
+          "type": "object",
+          "minProperties": 1,
+          "additionalProperties": { "$ref": "#/definitions/resource" }
         },
-        "resource": {
-            "type": "object",
-            "properties": {
-                "type": { "enum": [ "object" ] },
-                "description": { "$ref": "#/definitions/description" },
-                "required": { "$ref": "#/definitions/stringArrayEmptyOkay" },
-                "properties": {
-                    "type": "object",
-                    "minProperties": 1,
-                    "properties": {
-                        "type": { },
-                        "apiVersion": { "$ref": "#/definitions/apiVersionProperty" },
-                        "resources": { "$ref": "#/definitions/childResources" },
-                        "properties": { }
-                    },
-                    "required": [ "apiVersion", "type" ],
-                    "additionalProperties": { "$ref": "#/definitions/resourceProperty" }
-                }
-            },
-            "required": [ "description", "properties" ]
```

Comment:
...\azure-resource-manager-schemas\schemas\2015-06-01\Microsoft.KeyVault.json has resources that don't have "properties" as a resource-level property. That made me remove "properties" from the required list.